From bf3144bacb01a577b71fe9194a0d9f309a2f20b5 Mon Sep 17 00:00:00 2001
From: =?utf8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Mon, 15 Dec 2014 17:27:09 +0100
Subject: [PATCH] Substitute the correct GTK+ version in pkgconfig file

This fixes the build of GTK+ 3 version of lxappearance-obconf.
---
 configure.ac            | 1 +
 data/lxappearance.pc.in | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index a91c34b..2ba5d34 100644
--- a/configure.ac
+++ b/configure.ac
@@ -88,6 +88,7 @@ gmodule_modules="gmodule-export-2.0"
 PKG_CHECK_MODULES(GMODULE, [$gmodule_modules])
 AC_SUBST(GMODULE_CFLAGS)
 AC_SUBST(GMODULE_LIBS)
+AC_SUBST(gtk_modules)
 
 # gio_modules="gthread-2.0 gio-unix-2.0 glib-2.0 >= 2.18.0"
 # PKG_CHECK_MODULES(GIO, [$gio_modules])
diff --git a/data/lxappearance.pc.in b/data/lxappearance.pc.in
index 9abd35e..e711a31 100644
--- a/data/lxappearance.pc.in
+++ b/data/lxappearance.pc.in
@@ -6,5 +6,5 @@ datadir=@datadir@
 
 Name: lxappearance
 Description: Tool used to customize look and feel of gtk+ applications.
-Requires: gtk+-2.0 >= 2.12
+Requires: @gtk_modules@
 Version: @VERSION@
-- 
2.1.4

From 9e6c275c9e2cca534ecde1affc277d9a75d482fd Mon Sep 17 00:00:00 2001
From: =?utf8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Thu, 18 Dec 2014 13:27:20 +0100
Subject: [PATCH] Fix About dialog for GTK+ 3

---
 data/ui/about.glade.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/ui/about.glade.in b/data/ui/about.glade.in
index e1b1975..04f09f8 100644
--- a/data/ui/about.glade.in
+++ b/data/ui/about.glade.in
@@ -4,9 +4,9 @@
     <property name="border_width">5</property>
     <property name="window_position">center-on-parent</property>
     <property name="type_hint">dialog</property>
-    <property name="has_separator">False</property>
     <property name="program_name">LXAppearance</property>
     <property name="version">@VERSION@</property>
+    <property name="logo_icon_name">preferences-desktop-theme</property>
     <property name="copyright" translatable="yes">Copyright (C) 2008-2014 LXDE Project</property>
     <property name="comments" translatable="yes">Customizes look and feel of your desktop</property>
     <property name="website">http://lxde.org/</property>
-- 
2.1.4

From 970561f286d66056e0deb7ce9d5e276b2fd0187a Mon Sep 17 00:00:00 2001
From: =?utf8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Tue, 2 Feb 2016 00:41:20 +0100
Subject: [PATCH] Fix demo area background for GTK+ 3

Without this, the whole area become black, depending on the actually selected theme.
---
 src/widget-theme.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/widget-theme.c b/src/widget-theme.c
index 0a17326..3331ae4 100644
--- a/src/widget-theme.c
+++ b/src/widget-theme.c
@@ -134,12 +134,17 @@ static void on_font_set(GtkFontButton* btn, gpointer user_data)
 void widget_theme_init(GtkBuilder* b)
 {
     GtkWidget* demo;
+    GtkWidget* demo_vbox;
     GdkColor black = {0, 0, 0, 0};
 
     demo = GTK_WIDGET(gtk_builder_get_object(b, "demo"));
+    demo_vbox = GTK_WIDGET(gtk_builder_get_object(b, "demo_vbox"));
     app.widget_theme_view = GTK_WIDGET(gtk_builder_get_object(b, "widget_theme_view"));
 
     gtk_widget_modify_bg(demo, GTK_STATE_NORMAL, &black);
+#if GTK_CHECK_VERSION(3, 0, 0)
+    gtk_style_context_add_class (gtk_widget_get_style_context (demo_vbox), GTK_STYLE_CLASS_BACKGROUND);
+#endif
 
     app.widget_theme_store = gtk_list_store_new(1, G_TYPE_STRING);
 
-- 
2.1.4

From 527f4216cdf0ad97e78d3c747fe72a22ea0d7f47 Mon Sep 17 00:00:00 2001
From: =?utf8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Sun, 2 Aug 2015 00:45:37 +0200
Subject: [PATCH] Add MATE to NotShowIn in desktop entries

---
 data/lxappearance.desktop.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/lxappearance.desktop.in b/data/lxappearance.desktop.in
index 52fa037..54fc0a2 100644
--- a/data/lxappearance.desktop.in
+++ b/data/lxappearance.desktop.in
@@ -6,6 +6,6 @@ _Comment=Customizes look and feel of your desktop and applications
 _Keywords=windows;preferences;settings;theme;style;appearance;
 Icon=preferences-desktop-theme
 Exec=lxappearance
-NotShowIn=GNOME;KDE;XFCE;
+NotShowIn=GNOME;KDE;XFCE;MATE;
 StartupNotify=true
 Categories=GTK;Settings;DesktopSettings;X-LXDE-Settings;
-- 
2.1.4

